src/components/navs/AppNav/AppNavNetworkSelect.vue

```diff
@@ -94,23 +93,19 @@ onMounted(async () => {
 });
 
 // WATCHERS
-watchEffect(() => {
-  if (
-    chainId.value &&
-    previousNetwork.value &&
-    previousNetwork.value !== chainId.value
-  ) {
-    const newNetwork = allNetworks.value.find(
-      n => Number(n.key) === chainId.value
-    );
-    if (newNetwork) {
-      document.body.style.display = 'none';
-      localStorage.setItem('networkId', chainId.value.toString());
-      setWindowLocation(getNetworkChangeUrl(newNetwork));
+watch(
+  () => chainId.value,
+  (newVal, oldVal) => {
+    if (newVal && oldVal && oldVal !== newVal && networkId.value !== newVal) {
+      const newNetwork = allNetworks.value.find(n => Number(n.key) === newVal);
+      if (newNetwork) {
+        document.body.style.display = 'none';
+        localStorage.setItem('networkId', newVal.toString());
+        setWindowLocation(getNetworkChangeUrl(newNetwork));
+      }
```

Comment:
Use naming to make conditional a little clearer.
```suggestion
  (newChainId, oldChainId) => {
    if (newChainId && oldChainId && oldChainId !== newChainId && networkId.value !== newChainId) {
      const newNetwork = allNetworks.value.find(n => Number(n.key) === newChainId);
      if (newNetwork) {
        document.body.style.display = 'none';
        localStorage.setItem('networkId', newChainId.toString());
        setWindowLocation(getNetworkChangeUrl(newNetwork));
      }
```